cli/cli_fileshare.go

```diff
@@ -299,6 +326,82 @@ func (c *cmd) FileshareCancel(ctx *cli.Context) error {
 	return nil
 }
 
+// FileshareClear rpc
+func (c *cmd) FileshareClear(ctx *cli.Context) error {
+	if ctx.NArg() < 1 {
+		return formatError(argsCountError(ctx))
+	}
+
+	var resp *pb.Error
+	var err error
+	var until = time.Now().Unix()
+
+	args := ctx.Args()
+	if args.Get(0) != "all" {
+		argsJoined := strings.Join(args.Slice(), " ")
+		ts, err := parseTimespan(argsJoined)
+		if err != nil {
+			return formatError(err)
+		}
+		until -= ts
+	}
+
+	resp, err = c.fileshareClient.PurgeTransfersUntil(context.Background(), &pb.PurgeTransfersUntilRequest{Until: until})
+	if err != nil {
+		return formatError(err)
+	}
+	if err := getFileshareResponseToError(resp); err != nil {
+		return formatError(err)
+	}
+
+	color.Green(MsgFileshareClearSuccess)
+	return nil
+}
+
+func parseTimespan(ts string) (int64, error) {
+	var timespan int64 = 0
+	matches := timespanRegexp.FindAllString(ts, -1)
+
+	if matches == nil {
+		return 0, fmt.Errorf("Time span parsing error: '%s'", ts)
+	}
+
+	for _, m := range matches {
```

Comment:
does it make sense to support multiple values for time?

Reply:
See the first answer.